api/types/filters/parse.go

```diff
@@ -323,6 +323,20 @@ func (args Args) WalkValues(field string, op func(value string) error) error {
 	return nil
 }
 
+// Clone returns a copy of args.
+func (args Args) Clone() (newArgs Args) {
+	newArgs.fields = make(map[string]map[string]bool, len(args.fields))
+	for k, m := range args.fields {
+		if m != nil {
```

Comment:
Why not clone key with nil key?

Reply:
@tonistiigi so that if `args["foo"] == nil`, then `newArgs["foo"]` also `== nil`. Otherwise `newArgs["foo"] = map[string]bool{}`

Reply:
for the record, tonis changed my mind, because i didn't think of key presence as can be tested with `_, ok := m["foo"]`